### PR TITLE
route match: Fix runtime update bug and remove deprecated runtime flag

### PR DESCRIPTION
**Pull request by tonya11en**
*Description*:
This patch removes the `runtime` field from the `RouteMatch` message in route.proto, fixing #4623. In addition, the bug described in #4607 is fixed.

*Risk Level*:
Medium

*Testing*:
Enhanced unit tests to test for the scenario in #4607. Tests that verify behavior around the now removed `runtime` field have been removed.

*Docs Changes*:
n/a

*Release Notes*:
Updated.

### CODE REVIEW

File: docs/root/api-v1/route_config/route.rst [hunk 1]
```diff
@@ -236,16 +236,19 @@ include_vh_rate_limits
 Runtime
 -------
 
-A :ref:`runtime <arch_overview_runtime>` route configuration can be used to roll out route changes
+A :ref:`runtime_fraction <arch_overview_runtime>` route configuration can be used to roll out route changes
```
Comment (mattklein123) [inline docs/root/api-v1/route_config/route.rst]: I don't think the v1 config changed? Revert this file?

File: docs/root/intro/version_history.rst [hunk 2]
```diff
@@ -3,6 +3,7 @@ Version history
 
 1.9.0 (pending)
 ===============
+* router: Fixed bug related to runtime variable updates.
```
Comment (mattklein123) [inline docs/root/intro/version_history.rst]: Typically we don't release note bug fixes so I would remove.

File: source/common/router/config_impl.cc [hunk 3]
```diff
@@ -418,38 +417,36 @@ void RouteEntryImplBase::finalizeResponseHeaders(
   vhost_.globalRouteConfig().responseHeaderParser().evaluateHeaders(headers, request_info);
 }
 
+bool RouteEntryImplBase::evaluateRuntimeKeys(const RuntimeData& runtime_data,
+                                             const uint64_t random_value) const {
+  envoy::type::FractionalPercent fractional_percent;
+  const std::string& fraction_yaml = loader_.snapshot().get(runtime_data.fractional_runtime_key);
+
+  try {
+    MessageUtil::loadFromYamlAndValidate(fraction_yaml, fractional_percent);
```
Comment (mattklein123) [inline source/common/router/config_impl.cc]: This is in the data plane path, right? I think this is definitely not what we want. We need to store the key name, etc. and then do the runtime lookup?

**Comment by tonya11en** [comment]
I'm going to fix the bug and remove the deprecated field in two different PRs.

**Pull request closed without merging**